### ISSUE TITLE
(Orion) Dojo build for UI plugins doesn't seem quite right

### ISSUE THREAD

**Issue by aerwin** [maqetta/maqetta#2954]
Updated summary from 08-16-2012:

It's hard to call any of this regression since I don't think we've ever had a Dojo build in Orion. And, I'm not sure there was ever an official build of the standalone driver that contained my 6a1ab9931e7065c1c8554941590c214bd39f1343 from 14 days ago that re-enabled Dojo builds on the plug-ins (which had become broken with dojo 1.8.0rc1). At least I know we never did a test matrix on it.

But, I happen to have the standalone driver I built locally from master after merging 6a1ab9931e7065c1c8554941590c214bd39f1343, and notice some differences between that and Orion:
- UI plugins do not have minified JS in Orion (but do in my standalone driver).
- while an apples and oranges comparison in many ways, there are about 50 more requests dropping GridX on a page in the Orion build on maqetta-test than in my local standalone build.

**Comment by jhpedemonte** [comment]
@aerwin: Dojo is properly built on latest builds.  It sounds like you are saying that other areas of the build have regressed.  Is that correct?

If so, the title & summary should be updated to describe the exact areas that have regressed.

**Comment by aerwin** [comment]
@jhpedemonte 

I stand corrected... I downloaded the zip for the orion build and you are right that the UI plug-ins contain built Dojo. However, it seems like something is not quite right.

It's hard to call any of this regression since I don't think we've ever had a Dojo build in Orion. And, I'm not sure there was ever an official build of the standalone driver that contained my 6a1ab9931e7065c1c8554941590c214bd39f1343 from 14 days ago that re-enabled Dojo builds on the plug-ins (which had become broken with dojo 1.8.0rc1). At least I know we never did a test matrix on it.

But, I happen to have the standalone driver I built locally from master after merging 6a1ab9931e7065c1c8554941590c214bd39f1343, and notice some differences between that and Orion:
- UI plugins do not have minified JS in Orion (but do in my standalone driver).
- while an apples and oranges comparison in many ways, there are about 50 more requests dropping GridX on a page in the Orion build on maqetta-test than in my local standalone build.

I will update the title and placing above info in the summary.

**Comment by aerwin** [comment]
FYI... I went to lab9:50010 to try a scenario for @peller this morning, and the Dojo build for plugins looks more like I'd expect... e.g., at least the parts are minified.

**Comment by JonFerraiolo** [comment]
-> @jhpedemonte - note that with today's build on maqetta-test, it doesn't look like gridx is getting built

**Comment by jhpedemonte** [comment]
I have no idea how the GridX build works.  @childsb, @peller?

**Comment by jhpedemonte** [comment]
I noticed this in the build logs:

```
post.gather.bin.parts:
     [echo] Post gather for "gridx", version="1_0_0" ...
     [echo] Build gridx=true

davinciPlugin.build:
     [echo] Doing gridx build

backup.davinciPlugin.dojoDir:
     [copy] Copying 861 files to /var/lib/jenkins/jobs/maqetta-orion/workspace/I201208221522/plugins/davinci.gridx/davinci.gridx_1.0.0.201208221614
     [copy] Copying 1299 files to /var/lib/jenkins/jobs/maqetta-orion/workspace/I201208221522/plugins/davinci.gridx/davinci.gridx_1.0.0.201208221614
     [copy] Copying 342 files to /var/lib/jenkins/jobs/maqetta-orion/workspace/I201208221522/plugins/davinci.gridx/davinci.gridx_1.0.0.201208221614
     [copy] Copying 22 files to /var/lib/jenkins/jobs/maqetta-orion/workspace/I201208221522/plugins/davinci.gridx/davinci.gridx_1.0.0.201208221614/WebContent/maq-metadata-gridx
     [java] Java Result: 1
     [echo] caching gridx build...
    [mkdir] Created dir: /var/lib/jenkins/jobs/maqetta-orion/workspace/I201208221522/dojo_cache/gridx-1_0_0

cache.davinciPlugin.library:
     [copy] Copying 550 files to /var/lib/jenkins/jobs/maqetta-orion/workspace/I201208221522/dojo_cache/gridx-1_0_0

cache.davinciPlugin.metadata:
     [copy] Copying 22 files to /var/lib/jenkins/jobs/maqetta-orion/workspace/I201208221522/dojo_cache/gridx-1_0_0/metadata
   [delete] Deleting directory /var/lib/jenkins/jobs/maqetta-orion/workspace/I201208221522/plugins/davinci.gridx/davinci.gridx_1.0.0.201208221614/WebContent/release
   [delete] Deleting directory /var/lib/jenkins/jobs/maqetta-orion/workspace/I201208221522/plugins/davinci.gridx/davinci.gridx_1.0.0.201208221614/WebContent/util
   [delete] Deleting directory /var/lib/jenkins/jobs/maqetta-orion/workspace/I201208221522/plugins/davinci.gridx/davinci.gridx_1.0.0.201208221614/WebContent/dojo
   [delete] Deleting directory /var/lib/jenkins/jobs/maqetta-orion/workspace/I201208221522/plugins/davinci.gridx/davinci.gridx_1.0.0.201208221614/WebContent/dijit
   [delete] Deleting directory /var/lib/jenkins/jobs/maqetta-orion/workspace/I201208221522/plugins/davinci.gridx/davinci.gridx_1.0.0.201208221614/WebContent/maq-metadata-gridx
```

Notice the `Java Result: 1` line.

**Comment by aerwin** [comment]
Is the @dot.log for the plugin available?

**Comment by jhpedemonte** [comment]
[@dot.log](http://pastebin.com/d0v8isC4) for the GridX plugin build.

There are some warnings, but the first errors are this:

```
error(311) Missing dependency. module: maq-metadata-gridx/gridx/GridHelper; dependency: maq-metadata-dojo/dojox/grid/DataGridHelper
error(311) Missing dependency. module: maq-metadata-gridx/gridx/GridHelper; dependency: maq-metadata-dojo/dijit/layout/LayoutContainerHelper
error(311) Missing dependency. module: maq-metadata-gridx/gridx/GridHelper; dependency: maq-metadata-dojo/dijit/HTMLSubElementHelper
```

**Comment by jhpedemonte** [comment]
The maqetta-base build, which is successful in building GridX, shows this:

```
post.gather.bin.parts:
     [echo] Post gather for "gridx", version="1_0_0" ...
     [echo] Build gridx=true

davinciPlugin.build:
     [echo] Prepping for Dojo build of 'gridx' plugin...

backup.davinciPlugin.dojoDir:
     [copy] Copying 9412 files to /var/lib/jenkins/jobs/maqetta-base/workspace/tmp/eclipse/plugins/davinci.gridx_1.0.0.201208221651

copy.dojo18.metadata:
     [copy] Copying 398 files to /var/lib/jenkins/jobs/maqetta-base/workspace/tmp/eclipse/plugins/davinci.gridx_1.0.0.201208221651/WebContent/maq-metadata-dojo
     [copy] Copying 22 files to /var/lib/jenkins/jobs/maqetta-base/workspace/tmp/eclipse/plugins/davinci.gridx_1.0.0.201208221651/WebContent/maq-metadata-gridx
   [delete] Deleting directory /var/lib/jenkins/jobs/maqetta-base/workspace/tmp/eclipse/plugins/davinci.gridx_1.0.0.201208221651/WebContent/metadata
     [echo] Running Dojo build on 'gridx' plugin ...
     [echo] Cleaning up after Dojo build on 'gridx' plugin ...
    [mkdir] Created dir: /var/lib/jenkins/jobs/maqetta-base/workspace/dojo_cache/gridx_1_0_0

cache.davinciPlugin.library:
     [copy] Copying 932 files to /var/lib/jenkins/jobs/maqetta-base/workspace/dojo_cache/gridx_1_0_0

cache.davinciPlugin.metadata:
     [copy] Copying 32 files to /var/lib/jenkins/jobs/maqetta-base/workspace/dojo_cache/gridx_1_0_0/metadata
   [delete] Deleting directory /var/lib/jenkins/jobs/maqetta-base/workspace/tmp/eclipse/plugins/davinci.gridx_1.0.0.201208221651/WebContent/release
   [delete] Deleting directory /var/lib/jenkins/jobs/maqetta-base/workspace/tmp/eclipse/plugins/davinci.gridx_1.0.0.201208221651/WebContent/davinci
   [delete] Deleting directory /var/lib/jenkins/jobs/maqetta-base/workspace/tmp/eclipse/plugins/davinci.gridx_1.0.0.201208221651/WebContent/dijit
   [delete] Deleting directory /var/lib/jenkins/jobs/maqetta-base/workspace/tmp/eclipse/plugins/davinci.gridx_1.0.0.201208221651/WebContent/dojo
   [delete] Deleting directory /var/lib/jenkins/jobs/maqetta-base/workspace/tmp/eclipse/plugins/davinci.gridx_1.0.0.201208221651/WebContent/dojox
   [delete] Deleting directory /var/lib/jenkins/jobs/maqetta-base/workspace/tmp/eclipse/plugins/davinci.gridx_1.0.0.201208221651/WebContent/orion
   [delete] Deleting directory /var/lib/jenkins/jobs/maqetta-base/workspace/tmp/eclipse/plugins/davinci.gridx_1.0.0.201208221651/WebContent/preview
   [delete] Deleting directory /var/lib/jenkins/jobs/maqetta-base/workspace/tmp/eclipse/plugins/davinci.gridx_1.0.0.201208221651/WebContent/system
   [delete] Deleting directory /var/lib/jenkins/jobs/maqetta-base/workspace/tmp/eclipse/plugins/davinci.gridx_1.0.0.201208221651/WebContent/util
   [delete] Deleting directory /var/lib/jenkins/jobs/maqetta-base/workspace/tmp/eclipse/plugins/davinci.gridx_1.0.0.201208221651/WebContent/maq-metadata-gridx

delete.dojo18.metadata:
   [delete] Deleting directory /var/lib/jenkins/jobs/maqetta-base/workspace/tmp/eclipse/plugins/davinci.gridx_1.0.0.201208221651/WebContent/maq-metadata-dojo
```

Notice the bits about copying in `maq-metadata-dojo`.

---

Side note: the Dojo build in GridX is failing.  Why doesn't this result in a build failure?  The build just keeps going.

**Comment by aerwin** [comment]
> Side note: the Dojo build in GridX is failing. Why doesn't this result in a build failure? The build just keeps going.

That was going to be my next question... :)

What are the differences between the Orion build and the standalone build? The errors you point out are exactly the kinds of errors I fixed several weeks ago in 6a1ab9931e7065c1c8554941590c214bd39f1343 because the standalone build would die doing a dojo build.

So, aside from continuing to run are there other differences? Is the same widgetPluginDojoBuild.xml file being used in orion?

Are standalone builds still happening somewhere or is orion it now?

**Comment by aerwin** [comment]
> Are standalone builds still happening somewhere or is orion it now?

Guess you already answered that one with the output from the successful standalone build.

**Comment by jhpedemonte** [comment]
>  Is the same widgetPluginDojoBuild.xml file being used in orion?

So the Orion build seems to have duplicates (or mostly copies) of some build files, including **widgetPluginDojoBuild.xml** (_\_cough*\* @childsb _\_cough**).

Take a look at `releng/davinci.releng/orion/widgetPlugin/widgetPluginDojoBuild.xml` (and there's another similar one for the **internal** build).  I took a quick peak, but there are quite a few differences between the files.

@aerwin, since you know that file better, mind updating the Orion widgetPlugin/widgetPluginDojoBuild.xml files?

**Comment by aerwin** [comment]
Well, I can replace it the the official copy if that's what you want me to do.

But, I'd like to understand why we have duplicates? That's just asking for trouble. Seems like that's what needs to be fixed.

**Comment by jhpedemonte** [comment]
Yes, agreed about the duplicates.  I'll open a bug for this instance and others that I've run across.  But for M7, let's just keep the files where they are.

Be careful with doing a simple copy.  There may have been changes done specific to the Orion build, or to that location.

@childsb, care to chime in?

**Comment by aerwin** [comment]
> Be careful with doing a simple copy. There may have been changes done specific to the Orion build, or to that location.

Right... which is why I prefer not to do it without having a really good understanding of why it was changed in the first place. There was nothing platform specific in there.

**Comment by childsb** [comment]
It's duplicated because of gridx/CustomCallbacks.xml:

```
<!-- ===================================================================== -->
<!-- Custom targets called from a project's generated build.xml            -->
<!-- Set customBuildCallbacks=<path/to/this/file> in your build.properties.-->
<!-- ===================================================================== -->  
<project name="Build widget plugin" default="noDefault">
    <property environment="env"/>
    <import file="${env.relEngDir}/widgetPlugin/widgetPluginDojoBuild.xml"/>
</project>
```

the ${env.relEngDir} points to davinci.releng/orion/

and i didn't want to break the standalone build by changing it.

**Comment by aerwin** [comment]
Was it just duplicated or changed at all?

**Comment by childsb** [comment]
I had changed some of the if targets to this format:
`<target name="davinciPlugin.build" if="${run.dojo.build}" unless="${davinciPlugin.build.exists}">`

**Comment by aerwin** [comment]
And, did those changes make it back into the standalone file?

**Comment by aerwin** [comment]
And, @peller I know you were working on some  widgetPlugin/widgetPluginDojoBuild.xml changes today to delete test directories. Which version of the file have you been editing and testing with? :)

**Comment by peller** [comment]
@aerwin don't worry, I'm not planning on checking anything in today.

**Comment by peller** [comment]
@childsb, the orion ant file widgetPluginDojoBuild.xml was forked from the main file.  One of the changes was `failonerror="false"` for the Dojo build (<java> task)  That really ought to be 'true'.  We should probably review the diffs and try to reconcile and use a single file.

**Comment by jhpedemonte** [comment]
I did a straight copy of widgetPluginDojoBuild.xml into the `orion` dir and it worked locally. Committing now (and for internal build).  Will test again when new builds from _master_ are done.

**Comment by jhpedemonte** [comment]
OK, I confirmed that both the GridX code itself and the metadata helper files are getting built.

However, when testing Maqetta, I see unbuilt files returned for the GridX code (metadata helper files are correct).  Probably culprit is the 2nd `ajaxLibrary` element specified for "gridx-src":  Maqetta is getting confused and using the unbuilt files from "gridx-src" instead of "gridx" dir.  This will be fixed as part of #2988.

Closing this bug.
